server/src/main/java/io/crate/metadata/doc/DocTableInfo.java

```diff
@@ -1109,6 +1110,14 @@ private boolean addNewReferences(LongSupplier acquireOid,
                 }
                 addedColumn = true;
                 newReferences.put(newColumn, newRef.withOidAndPosition(acquireOid, positions::incrementAndGet));
+            } else if (exists.valueType().id() == NullArrayType.ID && newRef.valueType().id() == ArrayType.ID) {
+                // upgrade array_of_null to typed array
+                newReferences.put(newColumn, newRef);
```

Comment:
since we are adding new column to the schema, shouldn't we assign OID here?  ie use `newRef.withOidAndPosition`